### PR TITLE
Add support for InternalConstr

### PR DESCRIPTION
**Pull request by willzhang05**
Implemented the classes based on the schema in ISO 22901-1:2008 pg. 376-377, 394

### CODE REVIEW

**Comment by andlaus** [comment]
@willzhang05: what's the status here?

**Comment by willzhang05** [comment]
@andlaus Sorry, been busy with some other things, will try to address these later this week.

**Comment by andlaus** [comment]
> @andlaus Sorry, been busy with some other things, will try to address these later this week.

Great to read :) (I was just wondering if this was still on your radar...)

File: odxtools/internalconstr.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from xml.etree import ElementTree
+
+from .compumethods.limit import Limit
+from .odxtypes import DataType
+from .scaleconstr import ScaleConstr
+
+
+@dataclass
+class InternalConstr:
+    """This class represents a INTERNAL-CONSTR.
```
Comment (kayoub5) [inline odxtools/internalconstr.py:13]: this class now represents both internal and physical constr, so maybe update clasd name and dovs?

File: odxtools/scaleconstr.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    NOT_DEFINED = "NOT-DEFINED"
+    NOT_AVAILABLE = "NOT-AVAILABLE"
+
+
+@dataclass
+class ScaleConstr:
+    """This class represents a SCALE-CONSTR.
+    """
+
+    short_label: Optional[str]
```
Comment (kayoub5) [inline odxtools/scaleconstr.py:25]: there is a short label, but no long label?

File: odxtools/write_pdx_file.py [hunk 3]
```diff
@@ -119,6 +119,8 @@ def write_pdx_file(
         jinja_env.globals["make_bool_xml_attrib"] = make_bool_xml_attrib
 
         vars: Dict[str, Any] = {}
+        # set the index.xml short name tag to the filename without the file extension.
+        vars["ecu_name"] = os.path.splitext(os.path.basename(output_file_name))[0].replace(".", "_")
```
Comment (kayoub5) [inline odxtools/write_pdx_file.py:123]: those changes look unrelated to the rest of the PR, maybe move them to their own PR?

File: odxtools/scaleconstr.py (hunk 2 above)
Comment (uerg) [reply odxtools/scaleconstr.py:25]: yes, the XSD says only short label
image

File: odxtools/internalconstr.py (hunk 1 above)
Comment (andlaus) [reply odxtools/internalconstr.py:13]: this is exactly the same as in the XSD, so I think this is okay. (I agree that this should be done differently by the spec...)

File: odxtools/write_pdx_file.py (hunk 3 above)
Comment (andlaus) [reply odxtools/write_pdx_file.py:123]: also, this seems to be incorrect from a logical point of view: `output_file_name` is set by the `for` loop on line 100, so whatever it contains is just the first item of the last entry of `auxiliary_content_specifiers` (note that it also might not exist at all...)

Comment (andlaus) [reply odxtools/write_pdx_file.py:123]: further, the `ecu_name` variable does not seem to be used by the writing code, so these two lines should probably be removed...

File: odxtools/internalconstr.py (hunk 1 above)
Comment (kayoub5) [reply odxtools/internalconstr.py:13]: Class name wise, that's fine due to the spec, but the docs can still be updated, right?

Comment (andlaus) [reply odxtools/internalconstr.py:13]: true
```suggestion
    """This class represents a INTERNAL-CONSTR objects.

In DOPs, INTERNAL-CONSTR objects can be used to specify validity conditions for internal representations of quantities 
(INTERNAL-CONSTR) and for quantities in the physical domain (PHYS-CONSTR).
```

**Review by kayoub5 — APPROVED** [review]

**Comment by andlaus** [comment]
looks good; let's merge. thanks everybody!
